### PR TITLE
Refactor content.py

### PR DESCRIPTION
**Pull request by ikalnitsky**
It's really hard to explain what exactly was changed, so let's start
from the problem point of view. The problem is that the content.py was
ugly and had a bad design. There were a lot of unobvious things.

Here's a brief list of changes:

* the document class get rid of methods in favor of attributes
* the attributes are setted by __init__
* the destination attribute aren't a part of document interface now
* meta's attributes are document's attributes now
* the document ifself are sent to templates now

### CODE REVIEW

File: holocron/content.py [hunk 1]
```diff
@@ -53,258 +44,203 @@ class Document(metaclass=abc.ABCMeta):
...
-    def destination(self):
-        return os.path.join(self.app.conf['paths.output'], self.short_source)
+    """
 
-    @property
+    @cached_property
     def url(self):
+        # Since the Holocron's main idea is to preserve filesystem structure,
+        # a URL to the static document will be the same as a path to the
+        # document relative to the content directory.
```
Comment (PropheticBird) [inline holocron/content.py]: Can you rephrase this comment? Maybe split it in two sentences.

File: holocron/content.py [hunk 2]
```diff
@@ -53,258 +44,203 @@ class Document(metaclass=abc.ABCMeta):
...
-    the content folder.
-
-    For instance, a file::
+    A Page document implementation and representation.
 
-        /2014/01/01/birthday.rst
+    The Page document is a kind of Holocron's documents that converts some
+    markuped text document (i.g. markdown, restructuredtext) into HTML. The
+    conversion process is complex and includes both YAML-header parsing and
+    searching for valuable information in the content body (i.g. title).
```
Comment (PropheticBird) [inline holocron/content.py]: Here as well.

File: holocron/content.py [hunk 3]
```diff
@@ -53,258 +44,203 @@ class Document(metaclass=abc.ABCMeta):
...
-    an HTML document and saves the result into ``%filename%`` folder
-    with ``index.html`` filename, preserving the directory structure of
-    the content folder.
-
-    For instance, a file::
+    A Page document implementation and representation.
 
-        /2014/01/01/birthday.rst
+    The Page document is a kind of Holocron's documents that converts some
+    markuped text document (i.g. markdown, restructuredtext) into HTML. The
```
Comment (PropheticBird) [inline holocron/content.py]: Please correct `i.g.` to `e.g.` (exempli gratia).

File: holocron/content.py (hunk 1 above)
Comment (ikalnitsky) [inline holocron/content.py]: Ok.

File: holocron/content.py (hunk 3 above)
Comment (ikalnitsky) [inline holocron/content.py]: Ok.

File: holocron/content.py (hunk 2 above)
Comment (ikalnitsky) [inline holocron/content.py]: What do you mean by "Here as well."? Is this related to  *i.g.* mistake or to comment rephrasing?
